### PR TITLE
add basic context support

### PR DESCRIPTION
**Pull request by mpobrien**

### CODE REVIEW

File: otto.go [hunk 1]
```diff
@@ -303,12 +311,7 @@ func (self Otto) Run(src interface{}) (Value, error) {
 	return value, err
 }
 
-// Eval will do the same thing as Run, except without leaving the current scope.
-//
-// By staying in the same scope, the code evaluated has access to everything
-// already defined in the current stack frame. This is most useful in, for
-// example, a debugger call.
-func (self Otto) Eval(src interface{}) (Value, error) {
+func (self Otto) EvalWithContext(ctx goctx.Context, src interface{}) (Value, error) {
```
Comment (edaniels) [inline otto.go]: [q] Should this be passing the context?

File: otto.go [hunk 2]
```diff
@@ -552,6 +570,10 @@ func (self Otto) ContextSkip(limit int, skipNative bool) (ctx Context) {
 //      value, _ := vm.Call(`[ 1, 2, 3, undefined, 4 ].concat`, nil, 5, 6, 7, "abc")
 //
 func (self Otto) Call(source string, this interface{}, argumentList ...interface{}) (Value, error) {
+	return self.CallWithContext(goctx.Background(), source, this, argumentList)
+}
+
+func (self Otto) CallWithContext(ctx goctx.Context, source string, this interface{}, argumentList ...interface{}) (Value, error) {
```
Comment (edaniels) [inline otto.go]: [q] Same question for this regarding where the ctx goes

File: value.go [hunk 3]
```diff
@@ -118,6 +119,29 @@ func (value Value) Call(this Value, argumentList ...interface{}) (Value, error)
 	return result, err
 }
 
+func (value Value) CallWithContext(ctx context.Context, this Value, argumentList ...interface{}) (Value, error) {
+	result := Value{}
+	err := catchPanic(func() {
+		// FIXME
```
Comment (edaniels) [inline value.go]: [q] What is there to fix here or is it just copy paste from the other call?

File: value.go [hunk 4]
```diff
@@ -118,6 +119,29 @@ func (value Value) Call(this Value, argumentList ...interface{}) (Value, error)
 	return result, err
 }
 
+func (value Value) CallWithContext(ctx context.Context, this Value, argumentList ...interface{}) (Value, error) {
+	result := Value{}
```
Comment (edaniels) [inline value.go]: [nit] `var result Value`

File: value.go (hunk 3 above)
Comment (mpobrien) [reply value.go]: this is duplicate from the other call.

File: value.go (hunk 4 above)
Comment (mpobrien) [reply value.go]: ✅

File: otto.go (hunk 1 above)
Comment (mpobrien) [reply otto.go]: ✅

File: otto.go (hunk 2 above)
Comment (mpobrien) [reply otto.go]: ✅

**Comment by mpobrien** [comment]
@edaniels ready for a look. I updated Eval/EvalWithContext to be consistent with Call/CallWithContext and added tests.

**Pull request merged**